### DISCUSSION TITLE
Best practice for interfacing between COSMOS and cFS

### DISCUSSION THREAD

**Discussion by stanislaw** [nasa/cFS#326] [Q&A]
Hello,

I would like to ask the same question here on the cFS/discussions. The original question on COSMOS/discussions: [Best practice for interfacing between COSMOS and cFS](https://github.com/BallAerospace/COSMOS/discussions/1332).

---

We are setting up COSMOS, cFS and the CCDD tool in parallel to understand how these tools are supposed to work together.

So far, it looks like:

- COSMOS accepts two formats: its own text format or XTCE files.
- CCDD tool has its own internal format in the PostgreSQL database but it can export / import XTCE/EDS/JSON/CSV, but not the COSMOS text files.
- The cFS apps have their TM/TC definitions in the C header files.

Is our understanding right that a typical use case for these 3 tools includes CCDD tool being in the center of graph as a primary source of truth for data from which the XTCE or COSMOS files are generated as well as the cFS' header files? Can we assume that the XTCE import/export is the recommended exchange workflow between COSMOS and cFS (via CCDD)?

Are there alternative workflows? What is the best and practical approach for working with cFS and COSMOS?

If such a best practice exists, we would be happy to explore it, document it and send a patch to the COSMOS documentation.

Thanks.

**Comment by skliper** [comment]
The use case you describe matches what I've understood as the typical approach, but it has been a while since I worked that side of things.  cFS does also come with a set of EDS files (https://github.com/nasa/cFE/blob/main/modules/es/eds/cfe_es.xml for example) and there has been prototype code released open source that utilizes them directly at https://github.com/nasa/edslib.  @SpaceSteve121 may have some lesson's learned or experiences to share wrt CCDD (although I think his setup used a different ground station)?  There has been very recent work from the EDS side, which @jwilmot or @jphickey would have more recent status on.
